hip/CMakeLists.txt

```diff
@@ -265,8 +265,12 @@ if(GINKGO_HIP_PLATFORM MATCHES "${HIP_PLATFORM_AMD_REGEX}")
     get_target_property(HIP_LIBAMDHIP64_CONFIGURATIONS hip::amdhip64 IMPORTED_CONFIGURATIONS)
     if (CMAKE_BUILD_TYPE MATCHES "[Dd][Ee][Bb][Uu][Gg]" AND HIP_LIBAMDHIP64_CONFIGURATIONS MATCHES "[Dd][Ee][Bb][Uu][Gg]")
         get_target_property(HIP_LIBAMDHIP64_LIBRARIES hip::amdhip64 IMPORTED_LOCATION_DEBUG)
-    else()
+    elseif (CMAKE_BUILD_TYPE MATCHES "[Rr][Ee][Ll][Ee][Aa][Ss][Ee]" AND HIP_LIBAMDHIP64_CONFIGURATIONS MATCHES "[Rr][Ee][Ll][Ee][Aa][Ss][Ee]")
         get_target_property(HIP_LIBAMDHIP64_LIBRARIES hip::amdhip64 IMPORTED_LOCATION_RELEASE)
+    else ()
+        # Just get what is available
+        list(GET HIP_LIBAMDHIP64_CONFIGURATIONS 0 HIP_LIBAMDHIP64_FIRST_CONFIG)
+        get_target_property(HIP_LIBAMDHIP64_LIBRARIES hip::amdhip64 IMPORTED_LOCATION_${HIP_LIBAMDHIP64_FIRST_CONFIG})
```

Comment:
Well this is a fallback only, if both of the same type are available we get the correct one. I guess indeed it should work with `IMPORTED_LOCATION` as well but we could still have the same mismatch between debug/release in this case as well. This is also for things like `RelWithDebInfo` which `spack` uses quite often for example.

Reply:
you are right, it only takes the library and link it later, so ginkgo always link one of hip_libamdhip64 (maybe mismatch)
does hip contain several hip::amdhip64 release, debug, relwithdebinfo, ...?

Reply:
No it always contains one. If you use the package it's release, with spack it seems to be RelWithDebInfo. I changed this part so that it's cleaner, I will check if it still works with spack to make sure, but it should.